challenge_set_a_table/src/challenge_set_a_table_states/fetch_command.py

```diff
@@ -0,0 +1,87 @@
+#! /usr/bin/env python
+
+import rospy
+import smach
+import time
+import os
+import datetime
+import math
+
+import hmi
+
+from robot_smach_states import Initialize, Say
+from robot_smach_states.util.startup import startup
+
+def fetch(robot, time_out=15.0):
+
+    confirmed = False
+    robot.speech.speak('What can I do for you, master?')
+
+    # Listen for the new task
+    while not confirmed:
```

Comment:
This loop can now be infinite? Please put a reasonable max on it with a proper alternative